docs/src/domains/troubleshoot.md

```diff
@@ -25,25 +25,118 @@ main-abcd123.abcdefgh1234567.eu.platformsh.site has address 192.0.2.1
+It also assists when a CDN (Fastly, Cloudflare, ...) is used.
+It's good practice to check both the apex and the `www` domain to ensure that both point to your project.
+
+For more in-depth investigations, on the command line using macOS, Linux, or the Windows Subsystem for Linux, run the following command:
 
 ```bash
 curl -I -v  https://www.{{< variable "YOUR_DOMAIN" >}}
 ```
 
 The response should be long. Look for error messages.
 They're usually explicit enough.
-Often the problem is with a mismatch between the certificate and the domain name.
+Often the problem is caused by a mismatch between the certificate and the domain name or an expired [custom certificate](steps/tls.md).
+
+### Error provisioning certificates
+
+You may encounter Let's Encrypt certificates failing to provision after the build hook has completed:
+
+```bash
+Provisioning certificates
+  Validating 1 new domain
+  E: Error provisioning the new certificate, will retry in the background.
+  (Next refresh will be at 2022-02-13 14:29:22.860563+00:00.)
+  Environment certificates
+W: Missing certificate for domain a-new-and-really-awesome-feature-abc1234-defghijk56789.eu3.platformsh.site
+```
+
+The renewal may fail because of the 64-character limit Let's Encrypt places on URLs.
+If you have a branch with a long name, the environment URL is over this limit and the certificate is rejected.
+Shortening the branch name to fewer than 20 characters should solve the issue.
+
+Generated URLs for environments have the following pattern:
+
+```bash
+{{<variable "ENVIRONMENT" >}}-{{<variable "PROJECT_ID" >}}.{{<variable "REGION" >}}.platformsh.site
+```
+
+If you have a [default domain](../define-routes/_index.md#default) set up, the generated URL has the following pattern:
+
+```bash
+{{<variable "YOUR_DOMAIN" >}}.{{<variable "ENVIRONMENT" >}}-{{<variable "PROJECT_ID" >}}.{{<variable "REGION" >}}.platformsh.site
+```
+
+The generated URLs consist of:
+
+- `{{<variable "YOUR_DOMAIN" >}}` = the amount of characters your domain has
+- `{{<variable "ENVIRONMENT" >}}` = `{{<variable "BRANCH_NAME" >}}` + 7 character hash
+- `{{<variable "PROJECT_ID" >}}` = 13 characters
+- `{{<variable "REGION" >}}` = 2 to 4 characters, depending on the region
+- `platformsh.site` = 15 characters
+- extra characters like `.` and `-` = 4 to 5 characters, depending on if you have a default domain
+
+This leaves you with 21 to 23 characters for your branch name (`{{<variable "BRANCH_NAME" >}}`) without going over the 64-character limit,
+depending on the region.
+Since this pattern for generated URLs should remain similar even if it may change slightly,
+your branch names should be no more than 20 characters.
+
+### DNS Challenge
+
+To provide a valid SSL-certificate,
+Let's Encrypt needs to ensure that the requester is entitled to receive the SSL-certificate it asked for
+(usually through the presence of a specific token on the DNS zone of that domain).
```

Comment:
🛠️ I think you can split this long sentence into two to make it easier to read/ [avoid the use of parentheses](https://developers.google.com/style/parentheses).